src/transformStateWithClones.js

```diff
@@ -7,7 +7,39 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = actions;
+  let basis = { ...state };
+
+  for (const meaning in actions) {
+    const point = actions[meaning];
+
+    for (const keys in point) {
```

Comment:
You don't need to create an additional cycle.
You have to check `action.type` instead of this. You can use switch/case or just `if` block.
For example:
```
if (action.type === 'addProperties') {
  // use your code with `extraData`
}

if (action.type === 'removeProperties') {
  ...
}
```